content/en/tracing/trace_collection/custom_instrumentation/android/otel.md

```diff
@@ -0,0 +1,514 @@
+   * `TrackingConsent.NOT_GRANTED`: The SDK wipes all batched data and does not collect any future data.
+
+   Use the utility method `isInitialized` to check if the SDK is properly initialized:
+
+   ```kotlin
+    if (Datadog.isInitialized()) {
+        // your code here
+    }
+   ```
+   When writing your application, you can enable development logs by calling the `setVerbosity` method. All internal messages in the library with a priority equal to or higher than the provided level are then logged to Android's Logcat:
+   ```kotlin
+   Datadog.setVerbosity(Log.INFO)
+   ```
+
+3. Configure and enable Trace feature:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+val traceConfig = TraceConfiguration.Builder().build()
+Trace.enable(traceConfig)
+```
+{{% /tab %}}
+
+{{% tab "Java" %}}
+```java
+final TraceConfiguration traceConfig = TraceConfiguration.Builder().build();
+Trace.enable(traceConfig);
+```
+{{% /tab %}}
+{{< /tabs >}}    
+
+4. Datadog tracer implements the [Open Telemetry standard][18]. Create the `OtelTracerProvider` and register the `OpenTelemetrySdk` in `GlobalOpenTelemetry` as a singleton instance to be used in your application. We recommend for best practices to do this in your `onCreate()` method.
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+GlobalOpenTelemetry.set(object : OpenTelemetry {
+    private val tracerProvider = OtelTracerProvider.Builder()
+        .setService([BuildConfig.APPLICATION_ID])
+        .build()
+
+    override fun getTracerProvider(): TracerProvider {
+        return tracerProvider
+    }
+
+    override fun getPropagators(): ContextPropagators {
+        return ContextPropagators.noop()
+    }
+})
+// and later on if you want to access the tracer provider
+val tracerProvider = GlobalOpenTelemetry.get().getTracer(instrumentationName = "<instrumentation_name>")
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+
+{{% /tab %}} 
+{{% tab "Java" %}}
+```java
+GlobalOpenTelemetry.set(new OpenTelemetry() {
+    private final TracerProvider tracerProvider = new OtelTracerProvider.Builder()
+            .setService(BuildConfig.APPLICATION_ID)
+            .build();
+
+    @Override
+    public TracerProvider getTracerProvider() {
+        return tracerProvider;
+    }
+
+    @Override
+    public ContextPropagators getPropagators() {
+        return ContextPropagators.noop();
+    }
+};
+// and later on if you want to access the tracer provider
+final TracerProvider tracerProvider = GlobalOpenTelemetry.get().getTracer("<instrumentation_name>");       
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+**Note**: If you decide to set the `GlobalOpenTelemetry.set` API you need to make sure that this is only called one single time in your process otherwise OpenTelemetry will throw an exception. In case you want to avoid that you can just create a `TracerProvider` and use it as a singleton in your project.
```

Comment:
```suggestion
**Note**: Ensure `GlobalOpenTelemetry.set` API is only called once per process. Otherwise, you can create a `TracerProvider` and use it as a singleton in your project.
```